src/magic_folder/cli.py

```diff
@@ -136,6 +144,57 @@
 from .util.observer import ListenObserver
 
 
+class InitializeOptions(usage.Options):
+    """
+    Create and initialize a new Magic Folder daemon directory (which
+    will have no magic-folders in it; use "magic-folder create" for
+    that).
+    """
+
+    optParameters = [
+        ("config", "c", None, "A non-existant directory to contain config (default ~/.magic_folder)"),
+        ("listen-endpoint", "l", None, "A Twisted server string for our REST API (e.g. \"tcp:4321\")"),
+        ("tahoe-url", "t", None, "The HTTP URI for our Tahoe-LAFS client (e.g. \"http://localhost:3456\")"),
+    ]
+    description = (
+        "Initialize a new magic-folder daemon. A single daemon may run "
+        "any number of magic-folders (use \"magic-folder create\" to "
+        "create a new one."
+    )
+
+    def postOptions(self):
+        # defaults
+        if self['config'] is None:
+            self['config'] = expanduser("~/.magic_folder")
+
+        # required args
+        if self['listen-endpoint'] is None:
+            raise usage.UsageError("--listen-endpoint / -l is required")
+        if self['tahoe-url'] is None:
+            raise usage.UsageError("--tahoe-url / -t is required")
+
+        # validate
+        if exists(self['config']):
+            raise usage.UsageError("Directory '{}' already exists".format(self['config']))
```

Comment:
Dang I wish anyone at all was working on twisted.python.usage.  These are such basic, common checks; it would be great if you could just declare them.

Reply:
There is always `Click` :)